admin/projects.go

```diff
@@ -207,6 +205,88 @@ func (s *Service) UpdateProject(ctx context.Context, proj *database.Project, opt
 	return proj, nil
 }
 
+// UpdateProjectVariables updates a project's variables and runs reconcile on the deployments.
+func (s *Service) UpdateProjectVariables(ctx context.Context, project *database.Project, environment string, vars map[string]string, unsetVars []string, userID string) ([]*database.ProjectVariable, error) {
+	if len(vars) == 0 && len(unsetVars) == 0 {
+		return nil, nil
+	}
+	txCtx, tx, err := s.DB.NewTx(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
```

Comment:
Rollback should be safe to call after commit, so it can be defer called here for better safety: https://go.dev/doc/database/execute-transactions#example